ci/forgive-me.sh

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env bash
+
+set -x
+
+here="$(realpath "$(dirname "$0")")"
+source_dir="${here:?}/.."
+
+committish="${1:?}"
+tarballBaseName="${2:-solana-release}"
+remote=https://github.com/solana-labs/solana.git
+
+cleanup_tmp_source_dir() {
+  rm -rf "$tmp_source_dir"
+}
+
+set -e
+tmp_source_dir="$(mktemp --directory --tmpdir="$source_dir")"
+
+trap 'cleanup_tmp_source_dir' EXIT
+
+# Clone the specified committish
+git clone "$remote" "$tmp_source_dir"
```

Comment:
Is there any reason we have to clone all new? I think we can add another worktree like this:

```
# Create temporal source dir to build
git worktree add "$tmp_source_dir"
```

Reply:
Nope!  This was just a quick and dirty script to see if I could get what I wanted working (see script file name 😅).  Plenty of optimizations to be had here

Reply:
what a quick reply. :) I updated my comment by picking something from my attempt #12390. let's unify our work for the better. :)

Reply:
this way we can build off at arbitrary commit, might be useful for building with fast-changing release branches.